### PR TITLE
feat(hub-common): create an array of valid item types

### PR DESCRIPTION
**Pull request by sansth1010**
affects: @esri/hub-common

1. Description: ~~A constant array is created that contains all the valid item types.~~ Added a comment  linking to an issue to make `ItemType` enum up to date.

1. Instructions for testing:

1. Closes Issues: #6767 (if appropriate)

1. [ ] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [ ] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Comment by tomwayson** [comment]
I suspect that test coverage fail due to lake of 100% coverage.

I wonder if this is the best place to put this array. On the one hand I like that it's right next to the enum, on the other hand, presumably this file is for types. I'd almost rather have the array live in the server code for now and have a comment that links to the enum in Hub.js

File: packages/common/src/types.ts [hunk 1]
```diff
@@ -414,8 +414,151 @@ export enum FileExtension {
...
+  "Data Store",
+  "Big Data File Share",
+  "Hub Page",
+  "Hub Project",
+  "Code Attachment",
+  "Solution",
+  "Hub Initiative Template",
+  "Quickcapture Project",
+]
+
```
Comment (tomwayson) [inline packages/common/src/types.ts:558]: Once https://devtopia.esri.com/dc/hub-search/pull/159 merges, you can remove this from this PR and just leave the comment on the enum below.

Comment (sansth1010) [reply packages/common/src/types.ts]: Removed the array and just left the comment on enum.

**Review by sonofflynn89 — APPROVED** [review]
Woo hoo! 🎉

**Review by tomwayson — APPROVED** [review]
thanks @sansth1010

**Pull request merged**